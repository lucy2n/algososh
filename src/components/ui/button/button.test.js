import { Button } from "./button";
import React from "react";
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('The button without text is rendered without errors', () => {
    const buttonWithoutText = renderer.create(<Button />)
    expect(buttonWithoutText).toMatchSnapshot();
}); 

it('The button with text is rendered without errors', () => {
    const buttonWithText = renderer.create(<Button text="test"/>)
    expect(buttonWithText).toMatchSnapshot();
}); 

it('The locked button is rendered without errors', () => {
    const buttonDisabled = renderer.create(<Button disabled={true} />)
    expect(buttonDisabled).toMatchSnapshot();
}); 

it('The button with load indication is rendered without errors', () => {
    const buttonIsLoader = renderer.create(<Button isLoader={true} />)
    expect(buttonIsLoader).toMatchSnapshot();
}); 

it('Callback is working correctly', () => {
    const onClick = jest.fn();
    render(<Button text="test button" onClick={onClick}/>)
    const button = screen.getByText("test button");
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalled();
}); 
