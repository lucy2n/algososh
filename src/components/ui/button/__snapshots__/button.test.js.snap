// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The button with load indication is rendered without errors 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`The button with text is rendered without errors 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;

exports[`The button without text is rendered without errors 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`The locked button is rendered without errors 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
